### PR TITLE
fix: deprecation warning.

### PR DESCRIPTION
**Pull request by Traap**
Change vim.lsp.diagnostic.get_line_diagnostics() to vim.diagnostic.get().

### CODE REVIEW

**Comment by csaez** [comment]
@Traap thanks for the fix! :raised_hands:

**Comment by kosayoda** [comment]
Thanks for the PR! Can you confirm if this is backwards compatible? No worries if not.

**Comment by Traap** [comment]
@kosayoda - I build neovim daily on all systems I use.   I don't look back .... LOL

File: lua/nvim-lightbulb/init.lua [hunk 1]
```diff
@@ -259,7 +259,7 @@ NvimLightbulb.update_lightbulb = function(config)
     pcall(vim.b[bufnr].lightbulb_lsp_cancel)
     vim.b[bufnr].lightbulb_lsp_cancel = nil
   end
-  local context = { diagnostics = vim.lsp.diagnostic.get_line_diagnostics() }
+  local context = { diagnostics = vim.diagnostic.get() }
```
Comment (kbieganski) [inline lua/nvim-lightbulb/init.lua:262]:
```suggestion
  local context = { diagnostics = vim.diagnostic.get(0, { lnum = vim.fn.getcurpos()[2] }) }
```
Or something like that. `vim.diagnostic.get()` with no args [returns all diagnostics for all buffers](https://neovim.io/doc/user/diagnostic.html#vim.diagnostic.get()). Same below.

File: lua/nvim-lightbulb/init.lua [hunk 2]
```diff
@@ -393,7 +393,7 @@ NvimLightbulb.debug = function(config)
   end
 
   -- Send the LSP request
-  local context = { diagnostics = vim.lsp.diagnostic.get_line_diagnostics() }
+  local context = { diagnostics = vim.diagnostic.get() }
```
Comment (kbieganski) [inline lua/nvim-lightbulb/init.lua:396]:
```suggestion
  local context = { diagnostics = vim.diagnostic.get(0, { lnum = vim.fn.getcurpos()[2] }) }
```

**Review by kbieganski — COMMENTED** [review]
> Can you confirm if this is backwards compatible? 

Seems like this function has been there since [at least v0.6](https://github.com/neovim/neovim/blob/v0.6.0/runtime/lua/vim/diagnostic.lua#L694-L710).

File: lua/nvim-lightbulb/init.lua (hunk 2 above)
Comment (Traap) [reply lua/nvim-lightbulb/init.lua]: I add this change to my branch.  It does work the use cases I have used.

File: lua/nvim-lightbulb/init.lua (hunk 1 above)
Comment (Traap) [reply lua/nvim-lightbulb/init.lua]: See comment below.

**Comment by kosayoda** [comment]
Superseded by #69. Thank you very much regardless!

**Pull request closed without merging**